pkg/gameserversets/controller.go

```diff
@@ -640,3 +650,44 @@ func computeStatus(list []*agonesv1.GameServer) agonesv1.GameServerSetStatus {
 
 	return status
 }
+
+// aggregateCounters adds the contents of a CounterStatus map to an AggregatedCounterStatus map.
+func aggregateCounters(aggCounterStatus map[string]agonesv1.AggregatedCounterStatus, counterStatus map[string]agonesv1.CounterStatus) map[string]agonesv1.AggregatedCounterStatus {
+	if aggCounterStatus == nil {
+		aggCounterStatus = make(map[string]agonesv1.AggregatedCounterStatus)
+	}
+
+	for key, val := range counterStatus {
+		// If the Counter exists in both maps, aggregate the values.
+		if counter, ok := aggCounterStatus[key]; ok {
+			counter.Count += val.Count
+			counter.Capacity += val.Capacity
+			aggCounterStatus[key] = counter
+		} else {
+			aggCounterStatus[key] = agonesv1.AggregatedCounterStatus(*val.DeepCopy())
```

Comment:
nit: `agonesv1.AggregatedCounterStatus(*val.DeepCopy())` - this is different than how you handle it in line 718, what's the difference?

Reply:
We're casting type CounterStatus to type AggregatedCounterStatus, in 718 the inputs are both AggregatedCounterStatus, so we don't need to cast.